### PR TITLE
TF-1947 Swipe left in read item of ThreadView

### PR DESCRIPTION
**Pull request by sakriwassim**
https://github.com/linagora/tmail-flutter/assets/62904371/297aef1e-8815-4821-932b-989ba8a59503

https://github.com/linagora/tmail-flutter/assets/62904371/065feea5-3cb8-415f-96c6-bb1b53dc6667

we can use the solution but it is not like the ui 100%
the problem that we can not dismiss the tile by swiping 
I thing the first solution is butter .

https://github.com/linagora/tmail-flutter/assets/62904371/7d90f4fb-f1d6-41c0-991f-f73dd406b9d9

### CODE REVIEW

File: core/lib/presentation/resources/image_paths.dart [hunk 1]
```diff
@@ -190,6 +190,7 @@ class ImagePaths {
   String get icDelivering => _getImagePath('ic_delivering.svg');
   String get icError => _getImagePath('ic_error.svg');
   String get icConnectedInternet => _getImagePath('ic_connected_internet.svg');
+  String get icEmailOpen => _getImagePath('ic_email_open.svg');
```
Comment (nqhhdev) [inline core/lib/presentation/resources/image_paths.dart:193]: I don't see the uploaded .svg image file

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart [hunk 2]
```diff
@@ -25,6 +25,39 @@ mixin BaseEmailItemTile {
...
+  }) {
+    return Container(
+      width: iconSize ?? 56,
+      height: iconSize ?? 56,
+      alignment: Alignment.center,
+      decoration: BoxDecoration(
+          borderRadius: BorderRadius.circular((iconSize ?? 56) * 0.5),
+          border: Border.all(color: Colors.transparent),
+          color: AppColor.colorSpamReportBox
+      ),
```
Comment (nqhhdev) [inline lib/features/thread/presentation/mixin/base_email_item_tile.dart:40]:
```suggestion
      decoration: BoxDecoration(     
        borderRadius: BorderRadius.circular((iconSize ?? 56) * 0.5),
        border: Border.all(color: Colors.transparent),
        color: AppColor.colorSpamReportBox
      ),
```

Comment (nqhhdev) [reply lib/features/thread/presentation/mixin/base_email_item_tile.dart:40]: `borderRadius: BorderRadius.circular((iconSize ?? 56) * 0.5),`
Why did you use it?

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart [hunk 3]
```diff
@@ -25,6 +25,39 @@ mixin BaseEmailItemTile {
...
+      child: SvgPicture.asset(
+          imagePaths.icEmailOpen,
+          width: 24,
+          height: 24,
+          fit: BoxFit.fill),
+    );
```
Comment (nqhhdev) [inline lib/features/thread/presentation/mixin/base_email_item_tile.dart:46]: Convention

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart [hunk 4]
```diff
@@ -25,6 +25,39 @@ mixin BaseEmailItemTile {
...
+    return Text(
+        text??"",
+        maxLines: 1,
+        style: TextStyle(
+            fontSize: 15,
+            overflow: CommonTextStyle.defaultTextOverFlow,
+            color: AppColor.colorTextButton,
+            fontWeight: FontWeight.w600)
+    );
+  }
```
Comment (nqhhdev) [inline lib/features/thread/presentation/mixin/base_email_item_tile.dart:59]: Convention

File: lib/features/thread/presentation/widgets/dismissible_widget.dart [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+class DismissibleWidget<T> extends StatelessWidget with BaseEmailItemTile {
+  final T item;
+  final Widget child;
+  final DismissDirectionCallback onDismissed;
+
+  DismissibleWidget(
+      {super.key,
+      required this.item,
+      required this.onDismissed,
+      required this.child});
```
Comment (nqhhdev) [inline lib/features/thread/presentation/widgets/dismissible_widget.dart:14]: convention

File: lib/features/thread/presentation/widgets/dismissible_widget.dart [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        ),
+        child: ListTile(
+          leading: GestureDetector(
+            onTap: () {},
+            child: buildIconAvatarSwipe(),
+          ),
+          title: buildTitleTileSwipe("Marke as read"),
+        ),
+      );
+}
```
Comment (nqhhdev) [inline lib/features/thread/presentation/widgets/dismissible_widget.dart:38]: convention

**Comment by nqhhdev** [comment]
- Pls demo on IOS and Tablet, Ipad devices

File: lib/features/thread/presentation/widgets/dismissible_widget.dart [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        alignment: Alignment.centerLeft,
+        decoration: new BoxDecoration(
+          color: AppColor.colorItemRecipientSelected,
+        ),
+        child: ListTile(
+          leading: GestureDetector(
+            onTap: () {},
+            child: buildIconAvatarSwipe(),
+          ),
+          title: buildTitleTileSwipe("Marke as read"),
```
Comment (dab246) [inline lib/features/thread/presentation/widgets/dismissible_widget.dart:35]: Please add `string` to AppLocalizations to support multiple language

File: core/lib/presentation/resources/image_paths.dart (hunk 1 above)
Comment (dab246) [reply core/lib/presentation/resources/image_paths.dart:193]: Please check the icon is available in the `ImagePath` before adding a new one

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart (hunk 2 above)
Comment (sakriwassim) [reply lib/features/thread/presentation/mixin/base_email_item_tile.dart]: done

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart (hunk 3 above)
Comment (sakriwassim) [reply lib/features/thread/presentation/mixin/base_email_item_tile.dart]: done

File: lib/features/thread/presentation/widgets/dismissible_widget.dart (hunk 7 above)
Comment (sakriwassim) [reply lib/features/thread/presentation/widgets/dismissible_widget.dart]: done

File: lib/features/thread/presentation/widgets/dismissible_widget.dart (hunk 6 above)
Comment (sakriwassim) [reply lib/features/thread/presentation/widgets/dismissible_widget.dart]: done

File: lib/features/thread/presentation/mixin/base_email_item_tile.dart (hunk 4 above)
Comment (sakriwassim) [reply lib/features/thread/presentation/mixin/base_email_item_tile.dart]: done

File: lib/features/thread/presentation/thread_view.dart [hunk 8]
```diff
@@ -324,7 +325,9 @@ class ThreadView extends GetWidget<ThreadController>
           key: const PageStorageKey('list_presentation_email_in_threads'),
           itemExtent: _getItemExtent(context),
           itemCount: listPresentationEmail.length,
-          itemBuilder: (context, index) => Obx(() => _buildEmailItem(context, listPresentationEmail[index]))
+          itemBuilder: (context, index) => Obx(() => DismissibleWidget(textLeft:"Marke as read",item: listPresentationEmail[index],
```
Comment (hoangdat) [inline lib/features/thread/presentation/thread_view.dart:328]: i10n string please,

File: lib/features/thread/presentation/thread_view.dart [hunk 9]
```diff
@@ -324,7 +325,9 @@ class ThreadView extends GetWidget<ThreadController>
...
+          onDismissed: (DismissDirection direction) {  },
+          child: _buildEmailItem(context, listPresentationEmail[index])))
```
Comment (hoangdat) [inline lib/features/thread/presentation/thread_view.dart:330]: indent

Comment (sakriwassim) [reply lib/features/thread/presentation/thread_view.dart]: done

File: lib/features/thread/presentation/thread_view.dart (hunk 8 above)
Comment (sakriwassim) [reply lib/features/thread/presentation/thread_view.dart]: done

File: core/lib/presentation/resources/image_paths.dart (hunk 1 above)
Comment (sakriwassim) [reply core/lib/presentation/resources/image_paths.dart:193]: done

File: lib/features/thread/presentation/widgets/dismissible_widget.dart [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import 'package:core/presentation/extensions/color_extension.dart';
+import 'package:flutter/material.dart';
+import '../mixin/base_email_item_tile.dart';
```
[the remark this answers is not in the document]
Comment (sakriwassim) [reply lib/features/thread/presentation/widgets/dismissible_widget.dart]: done

**Comment by dab246** [comment]
@sakriwassim     **ATTENTION**!

1. Do the correct `DoD` in the ticket

Screenshot 2023-06-23 at 16 56 19


3. Squash `fixup` commits as a `fixup` commit before pushing a new commit.
4. Do `Resolve conversion` yourself when you're done with it.
5. Attach related `Issues` to PR (before the demo video) to make it easy for everyone to follow
6. Why doesn't it dismiss when no longer holding. Please find a way to solve it.


Screenshot 2023-06-23 at 17 11 03

**Pull request closed without merging**